src/lua/luaTask.c

```diff
@@ -57,78 +58,47 @@ static enum {
 
 static lua_State *g_lua;
 static xSemaphoreHandle xLuaLock;
-static unsigned int lastPointer;
 static size_t onTickSleepInterval;
+static size_t lua_mem_size;
 
-#ifdef ALLOC_DEBUG
-static int g_allocDebug = 0;
-#endif
-void * myAlloc (void *ud, void *ptr, size_t osize,size_t nsize)
+static void* myAlloc(void *ud, void *ptr, size_t osize, size_t nsize)
 {
-
-#ifdef ALLOC_DEBUG
-    if (g_allocDebug) {
-        SendString("myAlloc- ptr:");
-        SendUint((unsigned int)ptr);
-        SendString(" osize:");
-        SendInt(osize);
-        SendString(" nsize:");
-        SendInt(nsize);
-    }
-#endif
-
-    if (nsize == 0) {
-        portFree(ptr);
-#ifdef ALLOC_DEBUG
-        if (g_allocDebug) {
-            SendString(" (free)");
-            SendCrlf();
-        }
-#endif
-        return NULL;
-    } else {
-        void *newPtr;
-        if (osize != nsize) {
-            newPtr = portRealloc(ptr, nsize);
-        } else {
-            newPtr = ptr;
+        const int delta = nsize - osize;
+        const size_t new_lua_mem_size = lua_mem_size + delta;
+
+        if (nsize == 0) {
+                pr_debug_int_msg("[lua] RAM Freed: ", abs(delta));
```

Comment:
Macro to function call, but the function call will return quickly because not at trace level.  Also you are talking about a method that calls realloc most of the time.  I wouldn't be concerned about the performance of this line relative to others in this method.